apis/sls/v1alpha1/store_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+
+ Copyright 2021 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StoreSpec defines the desired state of SLS Store
+type StoreSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ForProvider field is where use set parameters for SLS Store
+	ForProvider StoreParameters `json:"forProvider"`
+}
+
+// StoreObservation is the representation of the current state that is observed.
+type StoreObservation struct {
+	// CreateTime is the time when the store was created
+	CreateTime uint32 `json:"createTime"`
+
+	// LastModifyTime is the time when the store was last modified
+	LastModifyTime uint32 `json:"lastModifyTime"`
+}
+
+// StoreStatus defines the observed state of SLS Store
+type StoreStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          StoreObservation `json:"atProvider,omitempty"`
+}
+
+// StoreParameters define the desired state of an SLS store.
+type StoreParameters struct {
+	// SLS project name
+	// +kubebuilder:validation:MinLength:=3
+	// +kubebuilder:validation:MaxLength:=63
+	ProjectName string `json:"projectName"`
```

Comment:
I believe you will want cross resource referencing implemented for this field to that folks are able to point to their `Project` resource in the cluster to populate the `ProjectName`. Let me know if you have any questions around implementation :)

Reply:
Correct. That's exactly what I need. I noticed a design doc on cross resource referencing https://github.com/crossplane/crossplane/blob/master/design/one-pager-cross-resource-referencing.md. A more straight example would be much appreciated.

Reply:
@hasheddan I thought it over. Resource SLS Store can also be independent of SLS Project. If the project doesn't exist, a `ProjectNotExist` error can occur https://www.alibabacloud.com/help/zh/doc-detail/29015.htm which is acceptable.